### PR TITLE
new drink: Jungle Juice

### PR DESCRIPTION
**Pull request by CrazyAmphibian**
[content]

## What this does
adds Jungle Juice, made with 3 parts berry juice, 2 parts vodka, and 1 part orange juice.
if jungle berries or poison berries are used, you get the "true" version, which will rapidly give toxin damage, up to a point, until it begins to heal toxin damage.

this also patches jungle berries to not be able to be seed extracted, since they gave you normal berry seeds anyways.

## Why it's good
more content is good, right? adds a bit of a ghetto chem to help with severe toxin damage in an interesting way.

## How it was tested
went into game and made the drink and then drunk it. it drinks.

## Changelog
:cl:
 * rscadd: added a new drink made with berry juice, vodka, and orange juice

### CODE REVIEW

File: code/modules/reagents/reagents/reagents_ethanol.dm [hunk 1]
```diff
@@ -2020,3 +2020,27 @@
...
+	glass_desc = "It's quite tart, with earthy undertones. Better hope the berries were safe."
+	nutriment_factor = 1.25 * REAGENTS_METABOLISM
+
+/datum/reagent/ethanol/drink/junglejuice/real/on_mob_life(var/mob/living/M)
+	if(..())
+		return 1
+	if(M.toxloss<15)
+		M.toxloss=min(M.toxloss+2.5,10)
+	else
+		M.toxloss = max(M.toxloss-1.5,10)
```
Comment (Eneocho) [inline code/modules/reagents/reagents/reagents_ethanol.dm]: Wouldn't this oscillate between healing the toxloss and causing it?

Comment (CrazyAmphibian) [reply code/modules/reagents/reagents/reagents_ethanol.dm]: oops, yeah. originality the threshold was 10, but i raised it to 15 and forgot to update the numbers in min and max

File: code/modules/reagents/Chemistry-Recipes.dm [hunk 2]
```diff
@@ -4430,5 +4430,27 @@
...
 	result_amount = 1
 
+/datum/chemical_reaction/junglejuice
+	name = "Jungle Juice"
+	id = JUNGLEJUICE
+	result = JUNGLEJUICE
+	required_reagents = list(BERRYJUICEJUNGLE = 1, SCREWDRIVERCOCKTAIL=1)
+	result_amount = 2
+
+/datum/chemical_reaction/junglejuice_fake
```
Comment (Inorien) [inline code/modules/reagents/Chemistry-Recipes.dm]: the PR in isolation is one thing but could you drop a comment in here just to explain why this exists, as well as at the defs for `junglejuice` and `junglejuice/real`?

also feel a bit weird about the only ingame way of knowing the difference between `JUNGLEJUICE` and `FAKEJUNGLEJUICE` being an extra line of examine text, seems pretty esoteric

**Review by Inorien — CHANGES_REQUESTED** [review]

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 2 above)
Comment (CrazyAmphibian) [reply code/modules/reagents/Chemistry-Recipes.dm]: to be completely honest the FAKE variant (created with normal berry juice) was only added because i wanted to avoid confusion that players would have over some berries being able to be turned into the drink and others not.
do with this information what you will.

Comment (Inorien) [reply code/modules/reagents/Chemistry-Recipes.dm]: just a line saying that in the code would be great, lest some developer reads this in the future and decides to "simplify" it

Comment (CrazyAmphibian) [reply code/modules/reagents/Chemistry-Recipes.dm]: added some comments, let me know if you think they're good

File: code/modules/reagents/reagents/reagents_ethanol.dm [hunk 3]
```diff
@@ -2032,7 +2032,7 @@
 	glass_desc = "It's quite tart, with earthy undertones."
 	nutriment_factor = 0.5 * REAGENTS_METABOLISM
 
-/datum/reagent/ethanol/drink/junglejuice/real
+/datum/reagent/ethanol/drink/junglejuice/real //the intended variant of jungle juice which you make from poisounous berries (or the ones found on jungle). this one includes the special effects, as well as higher nutriment content.
```
Comment (Inorien) [inline code/modules/reagents/reagents/reagents_ethanol.dm:2035]: poisounous

File: code/modules/reagents/Chemistry-Recipes.dm (hunk 2 above)
Comment (Inorien) [reply code/modules/reagents/Chemistry-Recipes.dm]: sure looks good its just to explain to future developers
we have some weird ancient workarounds to old shit in the code and are totally unexplained and its confusing, this clears it up

File: code/modules/reagents/reagents/reagents_ethanol.dm (hunk 3 above)
Comment (CrazyAmphibian) [reply code/modules/reagents/reagents/reagents_ethanol.dm:2035]: \>he merged it

**Comment by Eneocho** [comment]
Remember to add this to the Guide to Food and Drinks wiki page, with a caveat mentioning that there are two variants, depending on the juice used.

**Comment by Eneocho** [comment]
I just realized you never made a sprite for it.
